plugins/outputs/iotdb/iotdb.go

```diff
@@ -0,0 +1,287 @@
+package iotdb
+
+// iotdb.go
+
+import (
+	_ "embed"
+	"errors"
+	"fmt"
+	"math"
+
+	// Register IoTDB go client
+	"github.com/apache/iotdb-client-go/client"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/outputs"
+)
+
+// DO NOT REMOVE THE NEXT TWO LINES! This is required to embed the sampleConfig data.
+//go:embed sample.conf
+var sampleConfig string
+
+type IoTDB struct {
+	Host            string `toml:"host"`
+	Port            string `toml:"port"`
+	User            string `toml:"user"`
+	Password        string `toml:"password"`
+	Timeout         int    `toml:"timeout"`
+	ConvertUint64To string `toml:"convertUint64To"`
+	TimeStampUnit   string `toml:"timeStampUnit"`
+	TreateTagsAs    string `toml:"treateTagsAs"`
+	session         *client.Session
+
+	Log telegraf.Logger `toml:"-"`
+}
+
+type RecordsWithTags struct {
+	// IoTDB Records basic data struct
+	DeviceIDList     []string
+	MeasurementsList [][]string
+	ValuesList       [][]interface{}
+	DataTypesList    [][]client.TSDataType
+	TimestampList    []int64
+	// extra tags
+	TagsList [][]*telegraf.Tag
+}
+
+func (*IoTDB) SampleConfig() string {
+	return sampleConfig
+}
+
+// Init is for setup, and validating config.
+func (s *IoTDB) Init() error {
+	var errorMsg string
+	if s.Timeout < 0 {
+		errorMsg = fmt.Sprintf("IoTDB Config Warning: The value of 'timeout' is negative:%d. Now it's fixed to 0.", s.Timeout)
+		s.Log.Warnf(errorMsg)
+		s.Timeout = 0
+	}
+	if !(s.ConvertUint64To == "ToInt64" ||
+		s.ConvertUint64To == "ForceToInt64" ||
+		s.ConvertUint64To == "Text") {
+		errorMsg = fmt.Sprintf("IoTDB Config Warning: The value of 'ConvertUint64To' is invalid: %s. Now it's fixed to 'ToInt64'.", s.ConvertUint64To)
+		s.Log.Warnf(errorMsg)
+		s.ConvertUint64To = "ToInt64"
+	}
+	if !(s.TimeStampUnit == "second" ||
+		s.TimeStampUnit == "millisecond" ||
+		s.TimeStampUnit == "microsecond" ||
+		s.TimeStampUnit == "nanosecond") {
+		errorMsg = fmt.Sprintf("IoTDB Config Warning: The value of 'TimeStampUnit' is invalid: %s. Now it's fixed to 'nanosecond'.", s.TimeStampUnit)
+		s.Log.Warnf(errorMsg)
+		s.TimeStampUnit = "nanosecond"
+	}
+	if !(s.TreateTagsAs == "Measurements" || s.TreateTagsAs == "DeviceID_subtree") {
+		errorMsg = fmt.Sprintf("IoTDB Config Warning: The value of 'TreateTagsAs' is invalid: %s. Now it's fixed to 'Measurements'.", s.TreateTagsAs)
+		s.Log.Warnf(errorMsg)
+		s.TreateTagsAs = "Measurements"
+	}
+	s.Log.Info("IoTDB output plugin initialization completed.")
+	return nil
+}
+
+func (s *IoTDB) Connect() error {
+	// Make any connection required here
+	config := &client.Config{
+		Host:     s.Host,
+		Port:     s.Port,
+		UserName: s.User,
+		Password: s.Password,
+	}
+	var ss = client.NewSession(config)
+	s.session = &ss
+	if err := s.session.Open(false, s.Timeout); err != nil {
+		s.Log.Errorf("IoTDB Connect Error: Fail to connect host:'%s', port:'%s', err:%v", s.Host, s.Port, err)
+		return err
+	}
+
+	return nil
```

Comment:
Errors are logged automatically by Telegraf. Please also shorten the code...
```suggestion
	config := &client.Config{
		Host:     s.Host,
		Port:     s.Port,
		UserName: s.User,
		Password: s.Password,
	}
	s.session = &client.NewSession(config)
	if err := s.session.Open(false, s.Timeout); err != nil {
		return fmt.Errorf("connecting to %q failed: %w, s.Host, err)
	}

	return nil
```

Reply:
It's better now. Thank you for your careful review.